packages/x-data-grid/src/utils/domUtils.ts

```diff
@@ -71,6 +71,11 @@ export function findHeaderElementFromField(elem: Element, field: string): HTMLDi
   return elem.querySelector(`[data-field="${field}"]`)!;
 }
 
+export function getFieldsFromGroupHeaderElem(colCellEl: Element): string[] {
+  const found = colCellEl.getAttribute('data-fields')!.match(/^\|-(.+)-\|$/);
+  return found ? found[1].split('-|-') : [];
```

Comment:
I'd rather avoid regexp for better readability:
```suggestion
  const fieldsString = colCellEl.getAttribute('data-fields');
  return fieldsString?.startsWith('|-') ? fieldsString!.slice(2, -2).split('-|-') : [];
```

Reply:
Agree. Thanks 🙂